hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/cdc/CDCFileGroupIterator.scala

```diff
@@ -414,11 +385,10 @@ class CDCFileGroupIterator(split: HoodieCDCFileGroupSplit,
           }
           // load afterFileSlice to afterImageRecords
           if (currentCDCFileSplit.getAfterFileSlice.isPresent) {
-            val iter = loadFileSlice(currentCDCFileSplit.getAfterFileSlice.get())
+            val iter = loadFileSliceWithKeys(currentCDCFileSplit.getAfterFileSlice.get())
             afterImageRecords.clear()
-            iter.foreach { row =>
-              val key = getRecordKey(row)
-              afterImageRecords.put(key, row.copy())
+            iter.foreach { tuple =>
+              afterImageRecords.put(tuple._1, tuple._2.copy())
```

Comment:
nit: is `tuple._2.copy()` still needed as it goes through FGReader now?

Reply:
It's not, we will do the copy in FGReader now